### PR TITLE
graceful shutdown of websocket

### PR DESCRIPTION
**Pull request by valli0x**
checking for ctx.Done in a websockets subscription. It needed for graceful shutdown.

### CODE REVIEW

File: rpc/websockets.go [hunk 1]
```diff
@@ -411,9 +413,9 @@ type Header struct {
 }
 
 func (api *pubSubAPI) subscribeNewHeads(wsConn *wsConn, subID rpc.ID) (context.CancelFunc, error) {
-	ctx, cancel := context.WithCancel(context.Background())
+	_, cancel := context.WithCancel(context.Background())
```
Comment (mmsqe) [inline rpc/websockets.go]: This cancel function is obsolete as the context is not used anywhere.

Comment (valli0x) [reply rpc/websockets.go]: cancel func needed for eth_unsubscribe handling
I changed the code for this function

Comment (mmsqe) [reply rpc/websockets.go]: thanks, can you also sign the commits

Comment (valli0x) [reply rpc/websockets.go]: I signed the commits and added websocket error logging

**Review by mmsqe — APPROVED** [review]

**Pull request merged**